lib/foreman/renderer.rb

```diff
@@ -233,6 +277,11 @@ def load_template_vars
       @template_url = params['url']
     end
 
+    def global_setting(name, default = nil)
+      raise(FilteredGlobalSettingAccessed, _('Global setting "%s" is not accessible in safe-mode') % name) if Setting[:safemode_render] && !ALLOWED_GLOBAL_SETTINGS.include?(name.to_sym)
+      SETTINGS[name.to_sym] || default
```

Comment:
You should use `Setting`, not `SETTINGS`, the `SETTINGS` are just for things that come from settings file

Reply:
Also, you should check on `blank?`, instead of false value. Otherwise, `boolean` values, such as  `safemode_render`, would not work.